yledl/yledl.py

```diff
@@ -109,109 +109,12 @@ def open_config_file_and_remember_opened(filename: str, mode: str = 'r'):
     url_group = io_group.add_mutually_exclusive_group()
-    url_group.add_argument(
-        'url',
-        nargs='?',
-        type=str,
-        help='Address of an Areena, Elävä Arkisto, ' 'or Yle news web page',
-    )
-    url_group.add_argument(
-        '-i',
-        metavar='FILENAME',
-        dest='inputfile',
-        type=str,
-        help='Read input URLs to process from the named ' 'file, one URL per line',
-    )
-    io_group.add_argument(
-        '-o',
-        metavar='FILENAME',
-        dest='outputfile',
-        type=str,
-        help='Save stream to the named file',
-    )
-    io_group.add_argument(
-        '--output-template',
-        metavar='TEMPLATE',
-        default='${series_separator}${title}: ${episode_separator}${timestamp}',
-        help='Template for generating an output file name '
-        'when not using -o. Put the argument in single quotes: '
-        "--output-template '${title}'. "
-        'The template supports following substitutions: '
-        '${title} is replaced by the title of the episode, '
-        '${series} is the series title, '
-        '${series_separator} is the series title followed by ": ", '
-        '${season} is the string "Season XX" where XX is the season, '
-        '${episode} is the season and episode number ("S02E12"), '
-        '${episode_separator} is the season and episode followed by "-", '
-        '${timestamp} is stream publish timestamp ("2018-12-01T18:30"), '
-        '${date} is the stream publish date ("2018-12-01"), '
-        '${episode_or_date} is the same as "episode", or "date" if the episode '
-        'number is unavailable, '
-        '${program_id} is an unique ID, '
-        '$$ is an escape and will be replaced by a literal $. '
-        '/ specifies a subdirectory, but any / in the beginning is removed. '
-        'Everything else will appear as-is.',
-    )
-    io_group.add_argument(
-        '--output-na-placeholder',
-        metavar='PLACEHOLDER',
-        help='Placeholder value for unavailable meta fields '
-        'in output filename template '
-        '(default is an empty string)',
-    )
-    io_group.add_argument(
-        '--pipe',
-        action='store_true',
-        help='Dump stream to stdout for piping to media '
-        'player. E.g. "yle-dl --pipe URL | vlc -"',
-    )
-    io_group.add_argument(
-        '--destdir', metavar='DIR', type=str, help='Save files to DIR'
-    )
-    io_group.add_argument(
-        '--create-dirs',
-        action='store_true',
-        default=True,
-        help='Create directories automatically (this is the default)',
-    )
-    io_group.add_argument(
-        '--no-create-dirs',
-        action='store_false',
-        dest='create_dirs',
-        help='Stop if an output directory does not exist',
-    )
+    _add_url_groups_arguments(url_group)
+    _add_io_group_arguments(io_group)
     action_group = io_group.add_mutually_exclusive_group()
```

Comment:
Include everything from here until the `--xattrs` in the `_add_io_group_arguments` function.
(action_group and resume_group are subgroups of io_group so all of them should go together.)

Reply:
Before this PR I had a dilema - do the minimal change to remove the alert or handle the entire function.
I went with minimal impact but was not fully OK with that.
You solved my dilleman ;-)

I put each group in its own function (and resume_group was in the middle).
Is it OK?

Reply:
There seems to be circular dependencies that prevent code from running. For example, extractors.py imports stuff from areena_extractors.py, but areena_extractors.py imports something from extractors.py.

Maybe it's easier do this as two pull requests? Revert the extractor.py split here, then I can merge this without the split, and you can then do the splitting as a separate PR.

Reply:
You are correct.
I reverted the splitting in this PR.